### ISSUE TITLE
Docker system df does not show image size using devicemapper

### ISSUE THREAD

**Issue by frjaraur** [moby/moby#34927]
**Docker system df does not show image size using devicemapper**

When using devicemapper, docker system df does not show image layers space.
Not sure if issue is related to devicemapper.

**Steps to reproduce the issue:**
1. Create devicemapper thinpool using steps from https://docs.docker.com/engine/userguide/storagedriver/device-mapper-driver

2. Download a big image for seen results with df.
docker pull gitlab/gitlab-ce (for example)

3. Now use image listing and system df and compare results. 

**Describe the results you received:**
[root@vmcentos74 ~]# docker image ls
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
gitlab/gitlab-ce    latest              e471eb51ca7b        3 days ago          1.42GB

[root@vmcentos74 ~]# docker system df
TYPE                TOTAL               ACTIVE              SIZE                RECLAIMABLE
Images              1                   0                   0B                  0B
Containers          0                   0                   0B                  0B
Local Volumes       6                   0                   211.6MB             211.6MB (100%)
Build Cache                                                 0B                  0B

**Describe the results you expected:**
I expect to get same size because there is just one image.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
[root@vmcentos74 ~]# docker version
Client:
 Version:      17.07.0-ce
 API version:  1.31
 Go version:   go1.8.3
 Git commit:   8784753
 Built:        Tue Aug 29 17:42:01 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.07.0-ce
 API version:  1.31 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   8784753
 Built:        Tue Aug 29 17:43:23 2017
 OS/Arch:      linux/amd64
 Experimental: false

```

**Output of `docker info`:**

```
[root@vmcentos74 ~]# docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 1
Server Version: 17.07.0-ce
Storage Driver: devicemapper
 Pool Name: docker-thinpool
 Pool Blocksize: 524.3kB
 Base Device Size: 10.74GB
 Backing Filesystem: xfs
 Data file: 
 Metadata file: 
 Data Space Used: 1.865GB
 Data Space Total: 15.3GB
 Data Space Available: 13.43GB
 Metadata Space Used: 249.9kB
 Metadata Space Total: 159.4MB
 Metadata Space Available: 159.1MB
 Thin Pool Minimum Free Space: 1.529GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 0
 Library Version: 1.02.140-RHEL7 (2017-05-03)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 3addd840653146c90a254301d6c3a663c7fd6429
runc version: 2d41c047c83e09a6d61d464906feb2a2f3c52aa4
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-693.2.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 3.702GiB
Name: vmcentos74
ID: NZI2:YSQG:JQRB:UAUH:5WEQ:EJ6H:4JNP:3SRN:GIEO:7SXO:GWEO:SNGB
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Host Environment is created as VirtualBox machine.

**Comment by frjaraur** [comment]
Added extra info from image prune....

[root@vmcentos74 ~]# docker image prune -a
WARNING! This will remove all images without at least one container associated to them.
Are you sure you want to continue? [y/N] y
Deleted Images:
untagged: gitlab/gitlab-ce:latest
untagged: gitlab/gitlab-ce@sha256:d750327c99709ed9b3b567ddfd32373d041bd2d05e89d61f090c5e22a698a5bc
deleted: sha256:e471eb51ca7b7ace166faf6009241dfa486b13bc4897983e0cafe432829f3b48
deleted: sha256:c0afb8408e9aada48624cd208f42892117317cd203e1823b609637be32b5775f
deleted: sha256:eac9da95c8dff6ade92bb26b89fd574d37fdbf330f4c9b78f6615726909c2a2a
deleted: sha256:e8cbfe8db1fa480b031c5b48c9b02be6e6a8173eabcaf63f41920d091c643236
deleted: sha256:a587f19b1c0c8acb7ee10cddaec0f640c90cfee373a7a6970020d81142b03e55
deleted: sha256:ee084f1accf9fdc6093a1244b2eed7c5c18191ac33b943472a52ced78bcc486e
deleted: sha256:00bdcf3aa817a40388c218249a5a14e3878381a016fbd942fa81f0e39d89b03c
deleted: sha256:f8401e3776d248525490e0b8d11c47bef6030d654e2bf1e3d1042a94d28d9710
deleted: sha256:a13a3b4d67c74719752f113818dd06120458c20e4cda210af48e53ea8b7cdf66
deleted: sha256:2b43a0d4ab222c189664ef174b39f82c801d51808f61136f39ab3661ee408e25
deleted: sha256:11576f858b241ac8b03c42dd5b22cc4f02460afbfbad7e20b434c9a46a776ddf
deleted: sha256:8aa4fcad5eeb286fe9696898d988dc85503c6392d1a2bd9023911fb0d6d27081

Total reclaimed space: 1.42GB

Image size is successfuly reclaimed but docker system df didn't show it.

**Comment by thaJeztah** [comment]
I suspect this may not be related to devicemapper, but due to a bug in. 17.07 causing all sizes to be zero. This issue was fixed by https://github.com/moby/moby/pull/34826, and will be included in docker 17.09 (to be released soon)

**Comment by thaJeztah** [comment]
devicemapper was deprecated and removed. 
Let me close this ticket for now, as it looks like it went stale.
